### PR TITLE
Merge dev branch into master

### PR DESCRIPTION
**Pull request by Cyrof**
## Description:
This pull request merges changes from the `dev` branch into the `main` branch. It includes the following enhancements and additions: 

1. **Updated README.md file:**
   - Added github branch usage guide
2. **Added dataset folder, data exploration notebook, and seaborn package:**
   - Created a `dataset` fodler with `obesityData.csv` for data storage.
   - Added `data_exploration.ipynb` notebook for data exploration and visualisation.
   - Updated Pipfile to include seaborn package for enhanced data visualisation.

## Review: 
- **Reviewer:** @dixslyf 

## Additional Notes: 
This is the first merge to master branch

### CODE REVIEW

File: data_exploration.ipynb [hunk 1]
```diff
@@ -0,0 +1,1434 @@
...
+    "from sklearn.preprocessing import LabelEncoder\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading dataset\n",
+    "The dataset is loaded from a csv file from the `datasets` folder into a DataFrame using `pandas`. This allows for easy manipulation and analysis of data."
```
Comment (dixslyf) [inline data_exploration.ipynb:37]:
```suggestion
    "The dataset is loaded from a CSV file from the `datasets` folder into a `pandas` `DataFrame`."
```

Don't think we need that last sentence since it's fairly self-explanatory.

File: data_exploration.ipynb [hunk 2]
```diff
@@ -0,0 +1,1434 @@
...
+    "\n",
+    "ob_df = pd.read_csv(csv_path) # ob_df --> obesity dataframe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Explore the Dataset\n",
+    "Initial exploration involes checking the shape of the dataset, viewing the first few rows, and summarising basic statistics. This help in understanding the structure and basic characteristics of the data."
```
Comment (dixslyf) [inline data_exploration.ipynb:56]:
```suggestion
    "Initial exploration involves checking the shape of the dataset, viewing the first few rows, and summarising basic statistics. This helps us understand the structure and basic characteristics of the data."
```

File: data_exploration.ipynb [hunk 3]
```diff
@@ -0,0 +1,1434 @@
...
+       "max       3.000000     3.000000     2.000000  "
+      ]
+     },
+     "execution_count": 39,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "ob_df.describe()"
```
Comment (dixslyf) [inline data_exploration.ipynb:417]:
```suggestion
    "# Generate descriptive statistics for the data\n",
    "ob_df.describe()"
```

Could probably add a comment. (Hopefully I got the JSON right.)

File: data_exploration.ipynb [hunk 4]
```diff
@@ -0,0 +1,1434 @@
...
+    "plt.ylabel('Frequency')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    " ### 2. **Visualise the distribute of BMI (Body Mass Index)** <br>\n",
+    "Assuming BMI is calculated as weight in kg divided by height in meters squared (BMI = Weight / (Height/100)**2)"
```
Comment (dixslyf) [inline data_exploration.ipynb:1160]:
```suggestion
    "Assuming BMI is calculated as weight in kg divided by height in meters squared (`BMI = Weight / Height^2`)"
```

Our height data is already in metres.

File: data_exploration.ipynb [hunk 5]
```diff
@@ -0,0 +1,1434 @@
...
+    "plt.xlabel('Age')\n",
+    "plt.ylabel('Frequency')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    " ### 2. **Visualise the distribute of BMI (Body Mass Index)** <br>\n",
```
Comment (dixslyf) [inline data_exploration.ipynb:1159]:
```suggestion
    " ### 2. **Visualise the distribution of BMI (Body Mass Index)** <br>\n",
```

File: data_exploration.ipynb [hunk 6]
```diff
@@ -0,0 +1,1434 @@
...
+    "plt.xlabel('BMI')\n",
+    "plt.ylabel('Frequency')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3. **Boxplot of BMI by Gender**"
```
Comment (dixslyf) [inline data_exploration.ipynb:1194]:
```suggestion
    "### 3. **Box plot of BMI by Gender**"
```

File: data_exploration.ipynb [hunk 7]
```diff
@@ -0,0 +1,1434 @@
...
...h188THNJY5f0cZGWycC9kGenpvHr2AN+89i9Rgedq6tLzx6FK5QWgaEBFnj9WLQ+QkpSIhoYG+obGhdKrMMRGviExLgL3IhXl1/QMjHH08Odt2K2v9pyCiIl4Q2JcJJ7FlOu309+o315FFfU7MT6K12GhGJlYsmRcayb2q8yyCe158ej6F+mXUy6L5Wkri5Ysx1N1beWjUKVyDFA8MEhF3XFl+ph+DOhYi/FDOnIjT90BuHT2EP3b1+SPAS3Ztm4+qakpKnX8T5TBfM/NSEMDDaVJKy0dXTQ0JDx9cPOz8dPTM3gc9pzSJRVbmiQSCaVLluDeo8eF0kE2FsrAxFg2FpdKpVz66ybODvYMHj2Rxh160GvwCM5dvvaZlAQCMckiEMjZv38/RkZG8tCiRQuePn1KVlaWfJKkILp27cq2bdtITEwEICEhge3bt9OlSxe5zJMnTwqVVl5SU1OJj49XCgUtvwUw0pdNpCSmKG/hSUzJwki/cM/VAOqV1eLlRynhsaq3ApX2lhAem8XrCPVbhRITYpBKMzE2Vd5Xa2xqSXxspMo48bGRmOSRNzFTL6+KyI+yfbYHty6iUkhz+o5YhLNHEeaN6074+5cq4yRkp29sqvySYWRqKb+Xl08JsUilmRjljWNiSUKccpyLxzYxoktpRnQtw8Pb5+g+bDlaWrJBhZ6+Ib3/WMONC/sY3qkUI7qU4dHt83T7bQmampoqnx0fH49UKsXMzFzpurmZGTExMSrj5GXFqtVYWlhQKiBA5f20tDRWrFpNtarBGBoYFCrNvOSUgbxf1ozNLImLVb0EOT42Mp+8iamy/JFdq5BoalKjQeu/pVeOfVTaO+5z9i58ef4UH8OxXYupUKOFWl2unt5BYMX68v39/1/4t9g2/P0rTh7cTHDtH9XK5NThvLYzMrUiIS5CZZwkdfY2UW4TAoJ+oGWvqXQfvoaqDXtw88Jeti7+TSlOSPP+tO43i66/r6B42VrsWTOOi0fXy+9/r7YSYPvqaXj4lqRk2eoFyiVk54dpPvtZEBej2t5xsVGY5vGrkFs+Pi6alJQkDuxcQ4lSQQwePY9SFaoxf8pvPLyreJHtM2QSmRkZ9GsfQvcWFVmzaCIDhk7D1t75m+uYlwsnD6Cnb0jpIPX59fH9a44f2EK1Ok3VyuRGXpfy2r8Ae8bHRmJsVvjykp6Wyq71syl
```
Comment (dixslyf) [inline data_exploration.ipynb:1253]: This modifies our original `DataFrame`. We probably want to create a _view_ of the `DataFrame` and do the label encoding on that since we may want to use other encoding schemes like one-hot encoding for the actual training later.

We probably should enable copy-on write for Pandas as well (https://pandas.pydata.org/docs/dev/user_guide/copy_on_write.html).

Introduce a `ob_df_view = ob_df[:]`:

```suggestion
    "ob_df_view = ob_df[:]\n",
    "for column in ob_df_view.select_dtypes(include=['object']).columns:\n",
    "    label_encoders[column] = LabelEncoder()\n",
    "    ob_df_view[column] = label_encoders[column].fit_transform(ob_df[column])\n",
    "    \n",
    "# Calculate the correlation matrix\n",
    "correlation_matrix = ob_df_view.corr()\n",
```

Comment (dixslyf) [inline data_exploration.ipynb:1311]:
```suggestion
    "### 6. **Pairplot of a few selected features** <br>\n",
```

Comment (dixslyf) [inline data_exploration.ipynb:1281]: Should resolve this warning.

**Review by dixslyf — CHANGES_REQUESTED** [review]
Looking pretty good.

I think we have a bit too many graphs, so we might want to remove some of them in the future (doesn't have to be in this PR — we can decide which ones to remove later). However, we should at least have a graph that shows the relationship of some of the variables with the target variable (obesity level). I wonder if it's possible to colour code the points in the pairplot according to obesity level?

**Review by dixslyf — APPROVED** [review]
LGTM.
